src/NumericalAlgorithms/RootFinding/GslMultiRootFunctionFdf.cpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/RootFinding/GslMultiRootFunctionFdf.hpp"
+
+template <size_t Dim>
+int Gsl_MultiRootFunctionFdf_Wrapper_f(const gsl_vector* v, void* params,
+                                       gsl_vector* func) noexcept {
+  const GslMultiRootFunctionFdf<Dim>* fn =
+      static_cast<const GslMultiRootFunctionFdf<Dim>*>(params);
+  std::array<double, Dim> input_as_std_array;
+  for (size_t i = 0; i < Dim; i++) {
+    input_as_std_array[i] = (gsl_vector_get(v, i));
```

Comment:
Helper functions for array <-> gsl_vector conversion might be worth writing.  I'll leave that to your discretion.